Content.Shared/DeltaV/VendingMachines/ShopVendorComponent.cs

```diff
@@ -0,0 +1,96 @@
+using Robust.Shared.Audio;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+
+namespace Content.Shared.DeltaV.VendingMachines;
+
+/// <summary>
+/// A vending machine that sells items for a currency controlled by events.
+/// Does not need restocking.
+/// Another component must handle <see cref="ShopVendorBalanceEvent"/> and <see cref="ShopVendorPurchaseEvent"/> to work.
+/// </summary>
+[RegisterComponent, NetworkedComponent, Access(typeof(SharedShopVendorSystem))]
+[AutoGenerateComponentState, AutoGenerateComponentPause]
```

Comment:
\>adds the AutoGenerateComponentPause attribute
\>doesnt mark any fields as AutoPaused